### ISSUE TITLE
failed to remove dirs. failed to remove osd.4. failed to reweight osd.4 to 0.0: failed to crush reweight: ex

### ISSUE THREAD

**Issue by 13567436138** [rook/rook#1689]
```
lthrough": 0, "lsm_state": [0, 1, 0, 0, 0, 0, 0]}
2018-04-28 08:15:38.528989 I | osd7: 2018-04-28 08:15:38.528610 7f35e4d62700  4 rocksdb: EVENT_LOG_v1 {"time_micros": 1524903338528585, "job": 3, "event": "table_file_deletion", "file_number": 80}
2018-04-28 08:15:38.529001 I | osd7: 2018-04-28 08:15:38.528638 7f35e4d62700  4 rocksdb: EVENT_LOG_v1 {"time_micros": 1524903338528634, "job": 3, "event": "table_file_deletion", "file_number": 77}
2018-04-28 08:15:38.529012 I | osd7: 2018-04-28 08:15:38.528675 7f35e4d62700  4 rocksdb: EVENT_LOG_v1 {"time_micros": 1524903338528654, "job": 3, "event": "table_file_deletion", "file_number": 74}
2018-04-28 08:15:38.529023 I | osd7: 2018-04-28 08:15:38.528698 7f35e4d62700  4 rocksdb: EVENT_LOG_v1 {"time_micros": 1524903338528694, "job": 3, "event": "table_file_deletion", "file_number": 71}
2018-04-28 08:15:38.529035 I | osd7: 2018-04-28 08:15:38.528696 7f35f8bc9e00  0 osd.7 1024 crush map has features 2199057072128, adjusting msgr requires for clients
2018-04-28 08:15:38.529046 I | osd7: 2018-04-28 08:15:38.528754 7f35f8bc9e00  0 osd.7 1024 crush map has features 2199057072128 was 8705, adjusting msgr requires for mons
2018-04-28 08:15:38.529057 I | osd7: 2018-04-28 08:15:38.528755 7f35e4d62700  4 rocksdb: EVENT_LOG_v1 {"time_micros": 1524903338528714, "job": 3, "event": "table_file_deletion", "file_number": 69}
2018-04-28 08:15:38.529067 I | osd7: 2018-04-28 08:15:38.528764 7f35f8bc9e00  0 osd.7 1024 crush map has features 720578139436367872, adjusting msgr requires for osds
2018-04-28 08:15:38.529151 I | osd7: 2018-04-28 08:15:38.528852 7f35f8bc9e00  0 osd.7 1024 load_pgs
2018-04-28 08:15:38.529165 I | osd7: 2018-04-28 08:15:38.528867 7f35f8bc9e00  0 osd.7 1024 load_pgs opened 0 pgs
2018-04-28 08:15:38.529185 I | osd7: 2018-04-28 08:15:38.528874 7f35f8bc9e00  0 osd.7 1024 using weightedpriority op queue with priority op cut off at 64.
2018-04-28 08:15:38.530679 I | osd7: 2018-04-28 08:15:38.530568 7f35f8bc9e00 -1 osd.7 1024 log_to_monitors {default=true}
2018-04-28 08:15:38.562019 I | osd7: 2018-04-28 08:15:38.561427 7f35f8bc9e00  0 osd.7 1024 done with init, starting boot process
2018-04-28 08:15:39.109365 I | exec: Running command: ceph-osd --foreground --id=4 --cluster=rook --osd-data=/var/lib/rook/osd4 --conf=/var/lib/rook/osd4/rook.config --keyring=/var/lib/rook/osd4/keyring --osd-uuid=5dc469a5-46d1-4c64-aa3e-7df99243bdf4
2018-04-28 08:15:39.110238 I | proc: retry (total 0). started process [ceph-osd --foreground --id=4 --cluster=rook --osd-data=/var/lib/rook/osd4 --conf=/var/lib/rook/osd4/rook.config --keyring=/var/lib/rook/osd4/keyring --osd-uuid=5dc469a5-46d1-4c64-aa3e-7df99243bdf4]
2018-04-28 08:15:39.159017 I | exec: Error ENOENT: device 'osd.4' does not appear in the crush map
2018-04-28 08:15:39.159406 E | cephosd: failed to remove osd.4. failed to reweight osd.4 to 0.0: failed to crush reweight: exit status 2, Error ENOENT: device 'osd.4' does not appear in the crush map
. 
2018-04-28 08:15:39.165005 I | osd4: 2018-04-28 08:15:39.164754 7ff545601e00  0 ceph version 12.2.5 (cad919881333ac92274171586c827e01f554a70a) luminous (stable), process (unknown), pid 198
2018-04-28 08:15:39.167182 I | osd4: starting osd.4 at - osd_data /var/lib/rook/osd4 /var/lib/ceph/osd/rook-4/journal
2018-04-28 08:15:39.168275 I | osd4: 2018-04-28 08:15:39.168208 7ff545601e00  0 pidfile_write: ignore empty --pid-file
failed to remove dirs. failed to remove osd.4. failed to reweight osd.4 to 0.0: failed to crush reweight: exit status 2, Error ENOENT: device 'osd.4' does not appear in the crush map
```

**Comment by travisn** [comment]
@13567436138 do you have any more details on what the issue was? How many devices on the node? Was there a reboot? Had the cluster been healthy before this? Or is this the first run? Was it an existing cluster and you added a new node? etc

**Comment by travisn** [comment]
@13567436138 I believe this was fixed by #1892, but please reactivate if you see again on the latest builds.

**Comment by zhangxingdeppon** [comment]
hi,guy,did you reslove this problem.I also have this problem,can you help me?
